mantidproject/mantid
# Add multiple scattering correction to OSIRIS script

**Description of work**
This PR adds multiple scattering corrections to the OSIRIS Diffraction script.
**Purpose of work**

The aim of the work is to improve the current script's functionality.



**Summary of work**

Multiple scattering has been added to the OSIRIS script.

**Further detail of work**

`DiscusMultipleScatteringCorrection` has been added to the OSIRIS script. To enable it you need to set the `multiple_scattering` parameter to `True`. In addition, algorithm parameters `NeutronPathsSingle` and `NeutronPathsMultiple` has been added to the script. An example of how to use the multiple scattering scattering
```
osiris_focus.create_vanadium(run_number="120033")

osiris_focus.focus(run_number="120033",
    merge_drange=True,
    vanadium_normalisation=True,
    absorb_corrections=True,
    empty_can_subtraction_method="Simple",
    multiple_scattering=True
)
```

**To test:**
Make sure the following script works fine:
```
from mantid.simpleapi import *
import matplotlib.pyplot as plt
import numpy as np
from isis_powder.osiris import Osiris
from isis_powder import SampleDetails

osiris_focus = Osiris(user_name="Calib",
                  output_directory=r"\osiris\OsirisCalibrationFiles\output",
                  config_file=r"\osiris\OsirisCalibrationFiles\Config.yaml",
                  calibration_directory=r"\osiris\OsirisCalibrationFiles\Calibration_files",
                  calibration_mapping_file=r"\osiris\OsirisCalibrationFiles\Calibration_files\osiris_run_mapping.yaml"
)

sample_details = SampleDetails(radius=1.1, height=8, center=[0, 0, 0], shape='cylinder')
sample_details.set_material(chemical_formula='Cr2-Ga-N', number_density=10.0)
sample_details.set_container(
    chemical_formula='Al',
    mass_density=2.7,
    radius=1.2
)

osiris_focus.set_sample_details(sample=sample_details)

osiris_focus.create_vanadium(run_number="120033")

osiris_focus.focus(run_number="120033",
    absorb_corrections=True,
    empty_can_subtraction_method="Simple",
    paalman_pings_events_per_point=1000,
    sample_details=sample_details,
    multiple_scattering=True,
    simple_events_per_point=100,
    neutron_paths_single=5,
    neutron_paths_multiple=5,
)
```
Contact me for configuration files. Also, make sure you update the directories in the script with paths in your local machine.




Fixes #35430. 






---

#### Reviewer ####

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

#### Gatekeeper ####

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.
